app/controllers/DeploymentApprovalController.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.AppConfig
+import connectors.AuthConnector
+import model._
+import play.api.Logger
+import play.api.data.Form
+import play.api.Play.current
+import play.api.i18n.Messages.Implicits._
+import play.api.mvc.{Action, AnyContent, Request, Result}
+import services.DeploymentApprovalService
+import uk.gov.hmrc.play.frontend.controller.FrontendController
+import uk.gov.hmrc.play.http.HeaderCarrier
+import utils.{GatekeeperAuthProvider, GatekeeperAuthWrapper}
+import views.html.deploymentApproval.{deploymentApproval, deploymentReview}
+
+import scala.concurrent.Future
+
+object DeploymentApprovalController extends DeploymentApprovalController {
+  override val deploymentApprovalService = DeploymentApprovalService
+  override def authConnector = AuthConnector
+  override def authProvider = GatekeeperAuthProvider
+  override val appConfig = AppConfig
+}
+
+trait DeploymentApprovalController extends FrontendController with GatekeeperAuthWrapper {
+  val deploymentApprovalService: DeploymentApprovalService
+  implicit val appConfig: AppConfig
+
+  private def redirectToPendingPage() = {
```

Comment:
I don't like private/protected methods being near the top of a file, could you move the private/protected methods to one of:

* The bottom of the trait
* Inside a public method (if that is the only place the method is used)